validation/src/main/scala/hmda/validation/SubmissionLarStats.scala

```diff
@@ -2,28 +2,37 @@ package hmda.validation
 
 import akka.actor.{ ActorRef, ActorSystem, Props }
 import hmda.model.fi.SubmissionId
+import hmda.model.fi.lar.LoanApplicationRegister
 import hmda.persistence.messages.CommonMessages.{ Command, Event, GetState }
 import hmda.persistence.messages.events.processing.CommonHmdaValidatorEvents.LarValidated
 import hmda.persistence.model.HmdaPersistentActor
-import hmda.validation.ValidationStats.{ AddSubmissionSubmittedTotal, AddSubmissionValidationTotal, SubmissionStats }
+import hmda.validation.ValidationStats.{ AddSubmissionMacroStats, AddSubmissionSubmittedTotal }
+import hmda.validation.rules.lar.`macro`.Q071
 
 object SubmissionLarStats {
   val name = "SubmissionStats"
 
-  case class CountValidatedLarsInSubmission() extends Command
+  case class PersistStatsForMacroEdits() extends Command
   case class CountSubmittedLarsInSubmission() extends Command
-  case class SubmissionStatsUpdated(totalSubmitted: Int, totalValidated: Int) extends Event
+  case class SubmittedLarsUpdated(totalSubmitted: Int) extends Event
+  case class MacroStatsUpdated(totalValidated: Int, q071Total: Int, q071Sold: Int) extends Event
 
   def props(submissionId: SubmissionId): Props = Props(new SubmissionLarStats(submissionId))
 
   def createSubmissionStats(system: ActorSystem, submissionId: SubmissionId): ActorRef = {
     system.actorOf(SubmissionLarStats.props(submissionId))
   }
 
-  case class SubmissionLarStatsState(totalSubmitted: Int = 0, totalVerified: Int = 0) {
+  case class SubmissionLarStatsState(
+      totalSubmitted: Int = 0,
+      totalValidated: Int = 0,
+      q071Total: Int = 0,
+      q071SoldTotal: Int = 0
```

Comment:
Yeah... that's definitely the direction this is headed. I'm okay with it for now. I bet, after we add a few more edits, the right kind of refactor/optimization will be more obvious.